browser-signer-proxy: `BrowserSignerProxy::is_session_active` to return whether the session is active or not

TheAwiteb opened the pull request:
### Description

A function that returns `true` if the user is connected to the server and `false` if not

### Checklist

* [X] I followed the [contribution guidelines](https://github.com/rust-nostr/nostr/blob/master/CONTRIBUTING.md)
* [X] I ran `just precommit` or `just check` before committing
* [X] I updated the [CHANGELOG](https://github.com/rust-nostr/nostr/blob/master/CHANGELOG.md) (if applicable)

yukibtc reviewed (changes requested):
I think something should be changed here. For example, should be handle also the user disconnection? If yes, since the server accepts multiple connections, instead of using an `AtomicBool` I think we should use an `AtomicUsize`, counting the number of active sessions. If the sessions are 0, the user in not connected. What do you think?

TheAwiteb commented:
> I think something should be changed here. For example, should be handle also the user disconnection? If yes, since the server accepts multiple connections, instead of using an `AtomicBool` I think we should use an `AtomicUsize`, counting the number of active sessions. If the sessions are 0, the user in not connected. What do you think?

Agreed. The initial approach was simply to determine if the user connected to start using the proxy. It works for the CLI since it completes its task and exits, but for the GUI, there’s a need to track the user’s current connection

TheAwiteb commented:
Changes:
- Renamed the function from `user_connected` to `is_session_active` to better
reflect its purpose.
- Updated the function to return the real-time status of the session instead of
just a boolean for whether the session was open.

This is achieved by storing the timestamp of the last request to `/api/pending`,
which is the endpoint the client should poll for new requests from the native
app (polling interval set to 500ms). If no request is received for more than 2
seconds, the session is marked as inactive.
